### ISSUE TITLE
[feedback] c00-diamonds

### ISSUE THREAD

**Issue by zdelrosario** [llittle/data-science-work#1]
|Item      |Grade |
|:---------|:-----|
|Effort    |S     |
|Observed  |S     |
|Supported |S     |
|Styled    |U, 1  |

Comments

[1] In the future, use pipes to improve the readability of your code.

e.g. this

```
ggplot(data = diamonds) +
  geom_point(mapping = aes(x = carat, y = price))
```

becomes

```
diamonds %>%
  ggplot() +
  geom_point(mapping = aes(x = carat, y = price))
```\0

**Comment by llittle** [comment]
Question related to pipes:
In Section 4.4 of the style guide the example indicates to me that pipes are not necessary if it is a simple assignment where there are not several fiunctions chained together. 
So I don't understand why a pipe would be preferred for the `data = diamonds` part, especially if they are not preferred for the `geom_point` part (which is actually much more nested)?

**Comment by zdelrosario** [comment]
`ggplot` doesn't have the pipe operator (it was developed before `%>%` was 'discovered'), but you should think of `+` as ggplot's pipe. Thus we have several functions chained together.

Where this really pays off is when you start putting together both data processing and plotting. Compare:

```{r vis1}
mpg %>%
  pivot_longer(
    names_to = "Type",
    values_to = "Mileage",
    cols = c(cty, hwy)
  ) %>%
  select(Displacement = displ, Mileage, Type) %>%
  ggplot(aes(Displacement, Mileage, color = Type)) +
  geom_point()
```

with

```{r vis2}
ggplot(
  data = mpg %>%
  pivot_longer(
    names_to = "Type",
    values_to = "Mileage",
    cols = c(cty, hwy)
  ) %>%
  select(Displacement = displ, Mileage, Type),
  aes(Displacement, Mileage, color = Type)
) +
  geom_point()
```

- vis1 emphasizes the pre-processing before visualization
- vis2 embeds the pre-processing within the ggplot call

Getting in the habit of "putting your data first" will (in the long run) lead to more readable code.

I'm not sure I understand your comment about `geom_point()` being more nested....

**Comment by llittle** [comment]
Yeah that makes sense. It may be good to be explicit about this point in the rubric if you want to reinforce/grade on this convention because the RStudio primers assigned in the reading do not generally use pipes when passing data to ggplot. 

I am still figuring out how pipes work for more complex chains, so my comment about geom_point() may not actually make sense...

**Comment by zdelrosario** [comment]
That's a good point; I've added a **q** about this in the e-comm00-style based on your feedback. Thanks!
